core/src/test/java/io/questdb/std/datetime/microtime/TimestampsBruteForceTest.java

```diff
@@ -0,0 +1,148 @@
+
+package io.questdb.std.datetime.microtime;
+
+import org.junit.Test;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.*;
+import java.util.Locale;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+import static java.time.DayOfWeek.MONDAY;
+import static java.time.temporal.ChronoField.*;
+import static java.time.temporal.ChronoUnit.*;
+import static java.time.temporal.TemporalAdjusters.firstDayOfMonth;
+import static java.time.temporal.TemporalAdjusters.firstDayOfYear;
+import static org.junit.Assert.fail;
+
+public class TimestampsBruteForceTest {
+    private static final TemporalAdjuster TRUNCATE_TO_DECADE = (temporal -> {
+        // intentionally naive and different from the production impl
+        String yearString = String.valueOf(temporal.get(YEAR));
+        int shiftedYear = Integer.parseInt(replaceLastCharWithZero(yearString));
+        return temporal.with(YEAR, shiftedYear)
+                .with(DAY_OF_YEAR, 1)
+                .with(MICRO_OF_DAY, 0);
+    });
+
+    private static String replaceLastCharWithZero(String yearString) {
+        return yearString.substring(0, yearString.length() - 1) + "0";
+    }
+
+    private static final TemporalAdjuster TRUNCATE_TO_QUARTER = (temporal -> {
+        int month = temporal.get(MONTH_OF_YEAR);
+        int shiftedMonth;
+        // this is intentionally naive, to have the code as straightforward as possible
+        // and different from the production implementation under test
+        switch (month) {
+            case 1: case 2: case 3:
+                shiftedMonth = 1;
+                break;
+            case 4: case 5: case 6:
+                shiftedMonth = 4;
+                break;
+            case 7: case 8: case 9:
+                shiftedMonth = 7;
+                break;
+            case 10: case 11: case 12:
+                shiftedMonth = 10;
+                break;
+            default:
+                throw new AssertionError("More than 12 months in a year, huh? month: " + month);
+        }
+        return temporal.with(MONTH_OF_YEAR, shiftedMonth)
+                .with(DAY_OF_MONTH, 1)
+                .with(MICRO_OF_DAY, 0);
+    });
+
+    private static final TemporalAdjuster TRUNCATE_TO_CENTURY = (temporal -> {
+        int year = temporal.get(YEAR);
+        int yearRemainder = year % 100;
+        if (yearRemainder == 0) {
+            // 1900, 2000,...
+            year = year - 99;
+        } else {
+            year -= yearRemainder - 1;
+        }
+        return temporal.with(YEAR, year)
+                .with(DAY_OF_YEAR, 1)
+                .with(MICRO_OF_DAY, 0);
+    });
+
+
+    @Test
+    public void testFlooring() {
+        ZoneId utc = ZoneId.of("UTC");
+        ZonedDateTime current = ZonedDateTime.now(utc).withYear(1999);
+        ZonedDateTime deadline = current.plusYears(10);
```

Comment:
This is way too slow to afford CPU grill on every time we run test. I'd suggest:

- One day period is enough to test seconds, min, hours with existing increment.

- Then increment should be random number of days to test days, weeks, months, quarters for say 5 years.

- And increment of months is good enough to test decades, years, centuries for up to 10 years.

The change I suggest is to grow minimum increment after 1 day, 1 year, 5 years.

Reply:
It would be good if this test runs under 1s, it's hard to explain spending more time on each build to verify it.

Reply:
let me try it

Reply:
ok, I played with the parameters, the whole test class is running about 1.2s on my laptop. does it sound ok?